luxonis_ml/data/augmentations/utils.py

```diff
@@ -299,21 +386,30 @@ def post_transform_process(
         @type filter_kpts_by_bbox: bool
         @param filter_kpts_by_bbox: If True removes keypoint instances if its bounding
             box was removed.
-        @rtype: Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]
+        @rtype: Tuple[np.ndarray, Optional[np.ndarray], np.ndarray, np.ndarray]
         @return: Postprocessed annotations
         """
 
-        out_image = transformed_data["image"].astype(np.float32)
+        out_image = transformed_data["image"]
         ih, iw, _ = out_image.shape
         if not self.train_rgb:
             out_image = cv2.cvtColor(out_image, cv2.COLOR_RGB2BGR)
-
-        transformed_mask = transformed_data["mask"]
-        out_mask = np.zeros((ns, *transformed_mask.shape))
-        for key in np.unique(transformed_mask):
-            if key != 0:
-                out_mask[int(key) - 1, ...] = transformed_mask == key
-        out_mask[out_mask > 0] = 1
+        out_image = out_image.astype(np.float32)
+
+        out_mask = None
+        if return_mask:
+            transformed_mask = transformed_data.get("mask")
+            out_mask = (
+                np.zeros((ns, *transformed_mask.shape))
+                if transformed_mask is not None
```

Comment:
If `return_mask=True` in which case would we get that `transformed_mask=None` ?

Reply:
@JSabadin could you answer this?